scripts/prepro_feats.py

```diff
@@ -70,33 +68,51 @@ def main(params):
   if not os.path.isdir(dir_att):
     os.mkdir(dir_att)
 
-  for i,img in enumerate(imgs):
-    # load the image
-    I = skimage.io.imread(os.path.join(params['images_root'], img['filepath'], img['filename']))
-    # handle grayscale input images
-    if len(I.shape) == 2:
-      I = I[:,:,np.newaxis]
-      I = np.concatenate((I,I,I), axis=2)
-
-    I = I.astype('float32')/255.0
-    I = torch.from_numpy(I.transpose([2,0,1])).cuda()
-    I = Variable(preprocess(I), volatile=True)
-    tmp_fc, tmp_att = my_resnet(I, params['att_size'])
-    # write to pkl
-    np.save(os.path.join(dir_fc, str(img['cocoid'])), tmp_fc.data.cpu().float().numpy())
-    np.savez_compressed(os.path.join(dir_att, str(img['cocoid'])), feat=tmp_att.data.cpu().float().numpy())
-
-    if i % 1000 == 0:
-      print('processing %d/%d (%.2f%% done)' % (i, N, i*100.0/N))
+  fc_toc = {}
+  att_toc = {}
+  with h5py.File(os.path.join(dir_fc, 'feats_fc.h5')) as file_fc,\
+       h5py.File(os.path.join(dir_att, 'feats_att.h5')) as file_att:
+    for i, img in enumerate(imgs):
+      # load the image
+      I = skimage.io.imread(os.path.join(params['images_root'], img['filepath'], img['filename']))
+      # handle grayscale input images
+      if len(I.shape) == 2:
+        I = I[:,:,np.newaxis]
+        I = np.concatenate((I,I,I), axis=2)
+
+      I = I.astype('float32')/255.0
+      I = torch.from_numpy(I.transpose([2,0,1])).cuda()
+      I = Variable(preprocess(I), volatile=True)
+      tmp_fc, tmp_att = my_resnet(I, params['att_size'])
+      # write to hdf5
+
+      d_set_fc = file_fc.create_dataset(str(img['cocoid']), 
+        (2048,), dtype="float")
+      d_set_att = file_att.create_dataset(str(img['cocoid']), 
+        (params['att_size'], params['att_size'], 2048), dtype="float")
+
```

Comment:
What about:
d_set_fc = file_fc.create_dataset(str(img['cocoid']), 
        tmp_fc.shape, dtype="float")
d_set_att = file_att.create_dataset(str(img['cocoid']), 
        tmp_att.shape, dtype="float")

Reply:
No, this is necessary to create the hdf5 file. I created a dataset for each image with it's id.